Swift/Tests/CollectionTest.swift

```diff
@@ -626,6 +626,199 @@ class CollectionTest: CBLTestCase {
+            let _ = try col.save(document: mdoc)
+        }
+        expectError(domain: CBLErrorDomain, code: CBLError.notOpen) {
+            let _ = try col.save(document: mdoc, conflictHandler: { doc, doc2 in
+                return true
+            })
+        }
+        expectError(domain: CBLErrorDomain, code: CBLError.notOpen) {
+            let _ = try col.save(document: mdoc, concurrencyControl: .lastWriteWins)
+        }
+        
+        // delete functions
+        expectError(domain: CBLErrorDomain, code: CBLError.notOpen) {
+            let _ = try col.delete(document: doc)
+        }
+        expectError(domain: CBLErrorDomain, code: CBLError.notOpen) {
+            let _ = try col.delete(document: doc, concurrencyControl: .lastWriteWins)
+        }
+        
+        // purge functions
+        expectError(domain: CBLErrorDomain, code: CBLError.notOpen) {
+            let _ = try col.purge(document: doc)
+        }
+        expectError(domain: CBLErrorDomain, code: CBLError.notOpen) {
+            let _ = try col.purge(id: "doc2")
+        }
+        
+        // doc expiry
+        expectError(domain: CBLErrorDomain, code: CBLError.notOpen) {
+            let _ = try col.setDocumentExpiration(id: "doc6", expiration: Date())
+        }
+        expectError(domain: CBLErrorDomain, code: CBLError.notOpen) {
+            let _ = try col.getDocumentExpiration(id: "doc6")
+        }
+        
+        // indexes
+        let config = ValueIndexConfiguration.init(["firstName"])
+        expectError(domain: CBLErrorDomain, code: CBLError.notOpen) {
+            let _ = try col.createIndex(withName: "index1", config: config)
+        }
+        expectError(domain: CBLErrorDomain, code: CBLError.notOpen) {
+            let _ = try col.indexes()
+        }
+        expectError(domain: CBLErrorDomain, code: CBLError.notOpen) {
+            let _ = try col.deleteIndex(forName: "index2")
+        }
+    }
+    
+    // MARK: 8.7 Use Scope APIs on deleted/closed scenarios
+    
+    func testUseScopeWhenDatabaseIsClosed() throws {
+        try testUseInvalidScope() {
+            try self.db.close()
+        }
+    }
+    
+    func testUseScopeWhenDatabaseIsDeleted() throws {
+        try testUseInvalidScope() {
+            try self.db.delete()
+        }
+    }
+    
+    func testUseInvalidScope(_ onAction: () throws -> Void) throws {
+        let colA = try self.db.createCollection(name: "colA")
+        let scope = colA.scope
+        
+        try onAction()
+        
+        expectError(domain: CBLErrorDomain, code: CBLError.notOpen) {
+            let _ = try scope.collection(name: "colA")
+        }
+        expectError(domain: CBLErrorDomain, code: CBLError.notOpen) {
+            let _ = try scope.collections()
+        }
+    }
+    
+    // MARK: 8.8 Get Scopes/Collections
+    
+    func testGetScopesOrCollectionsWhenDatabaseIsClosed() throws {
+        try self.db.close()
```

Comment:
Does it need to call getScopesOrCollectionsTest()?

Reply:
OMG!  🤦‍♂️